pkg/errors/factory_test.go

```diff
@@ -154,14 +167,82 @@ func TestIsForbiddenError(t *testing.T) {
 		},
 	}
 	for _, tc := range cases {
-		tc := tc
 		t.Run(tc.Name, func(t *testing.T) {
 			t.Parallel()
-			assert.Equal(t, IsForbiddenError(tc.Error), tc.Forbidden)
+			assert.Equal(t, IsForbiddenError(GetError(tc.Error)), tc.Forbidden)
 		})
 	}
 }
 
 func TestNilError(t *testing.T) {
 	assert.NilError(t, GetError(nil), nil)
 }
+
+func TestIsInternalError(t *testing.T) {
+	cases := []struct {
+		Name     string
+		Error    error
+		Internal bool
+	}{
+		{
+			Name:     "internal error with connection refused",
+			Error:    api_errors.NewInternalError(fmt.Errorf("failed calling webhook \"webhook.serving.knative.dev\": Post \"https://webhook.knative-serving.svc:443/defaulting?timeout=10s\": dial tcp 10.96.27.233:443: connect: connection refused")),
+			Internal: true,
+		},
+		{
+			Name:     "internal error with context deadline exceeded",
+			Error:    api_errors.NewInternalError(fmt.Errorf("failed calling webhook \"webhook.serving.knative.dev\": Post https://webhook.knative-serving.svc:443/defaulting?timeout=10s: context deadline exceeded")),
+			Internal: true,
+		},
+		{
+			Name:     "internal error with i/o timeout",
+			Error:    api_errors.NewInternalError(fmt.Errorf("failed calling webhook \"webhook.serving.knative.dev\": Post https://webhook.knative-serving.svc:443/defaulting?timeout=10s: i/o timeout")),
+			Internal: true,
+		},
+		{
+			Name:     "not internal error",
+			Error:    mockErrType{},
+			Internal: false,
+		},
```

Comment:
What about unknown errors? Or the previous covers it?

Reply:
The previous function covers the non k8s api errors. This one covers a mock one which implements the k8s api interface.

Reply:
@maximilien any more comments to the above please?